src/Data/Promotion/Prelude.hs

```diff
@@ -42,6 +42,9 @@ module Data.Promotion.Prelude (
   -- * Promoted numbers
   module Data.Promotion.Prelude.Num,
 
+  -- * Promoted 'Show'
+  module Data.Promotion.Prelude.Show,
```

Comment:
A consequence of this, unfortunately, is that we have to add more `import Data.Singletons.Prelude.Show` statements to various test cases to work around #190. Oh well.